src/Microsoft.Azure.SignalR/EndpointProvider/ServiceEndpointProvider.cs

```diff
@@ -31,21 +34,31 @@ public ServiceEndpointProvider(ServiceEndpoint endpoint, TimeSpan? ttl = null)
 
             _endpoint = endpoint.Endpoint;
             _accessKey = endpoint.AccessKey;
+            _serviceOptions = serviceOptions;
+            _appName = appName;
 
             var port = endpoint.Port;
             var version = endpoint.Version;
 
             _generator = new DefaultServiceEndpointGenerator(_endpoint, _accessKey, version, port);
         }
 
+        internal string ApplicationName

```

Comment:
is private field enough?

Reply:
In theory IOptions.Value can change over time, so I evaluate each time.